local-cluster/src/local_cluster.rs

```diff
@@ -907,23 +907,23 @@ impl Cluster for LocalCluster {
         &mut self,
         pubkey: &Pubkey,
         cluster_validator_info: &mut ClusterValidatorInfo,
-    ) -> (Node, Option<ContactInfo>) {
+    ) -> (Node, Vec<ContactInfo>) {
         // Update the stored ContactInfo for this node
         let node = Node::new_localhost_with_pubkey(pubkey);
         cluster_validator_info.info.contact_info = node.info.clone();
         cluster_validator_info.config.rpc_addrs =
             Some((node.info.rpc().unwrap(), node.info.rpc_pubsub().unwrap()));
 
-        let entry_point_info = {
-            if pubkey == self.entry_point_info.pubkey() {
-                self.entry_point_info = node.info.clone();
-                None
-            } else {
-                Some(self.entry_point_info.clone())
-            }
-        };
+        let entry_point_infos: Vec<ContactInfo> = self
+            .validators
+            .values()
+            .map(|validator| validator.info.contact_info.clone())
+            .collect();
+        if entry_point_infos.is_empty() {
+            panic!("Validator has no alive entrypoints to rejoin cluster with");
```

Comment:
saw some tests are failing here, might want to set the entry point to the node itself in case no one is alive (or single node cluster).

Reply:
maybe this assertion is just not a good one then, since it's possible for a node to start up when everyone else is dead.

You're in a bind though if you restart by yourself with no entrypoints/outdated entrypoint and then the rest of the cluster boots up afterwards, you won't be able to be discovered by everybody else

Reply:
this should just work right - once the second validator starts up it will use the first guy as an entrypoint from the above collect? 
Also do we want to update the `LocalCluster` 's entrypoint? previous code was doing this:
```
                self.entry_point_info = node.info.clone();
```
we could do this on restart if the previous entry point is dead?

Reply:
yeah, that's a good catch, updating the entrypoint as was done previously